languages/unicode.lua

```diff
@@ -72,14 +72,13 @@ SILE.nodeMakers.unicode = pl.class({
         self:addToken(char, item)
         self:makeToken()
         self:makePenalty(0)
-      elseif self.lasttype and (self.thistype and thistype ~= lasttype and not self.isWordType[thistype]) then
+      elseif self.lasttype and self.thistype and thistype ~= self.lasttype and not self.isWordType[thistype] then
         self:makeToken()
         self:addToken(char, item)
       else
         self:letterspace()
         self:addToken(char, item)
       end
-      if not self.isWordType[thistype] then lasttype = chardata[cp] and chardata[cp].linebreak end
       self.lasttype = thistype
```

Comment:
@simoncozens If you have any idea what this bit of code was ever supposed to please jump in here. It's obviously been broken for a very long time. I traced it back far enough in history to know my recent refactoring isn't the problem, this has had several kinds of logic issues for a long time. The variable scope is so wrong that Lua would never pass the value anyway, and even if it did the next if statement would obliterate it. Given that none of the tests break no matter what I do to this I don't know that what I did here is actually a correct fix.

Reply:
Promoted to issue #699.